### ISSUE TITLE
Add State Design Pattern In Java

### ISSUE THREAD

**Issue by ighosh98** [OpenGenus/cosmos#5924]
**Details:**

**Details:**

**This is a(n):**
- [x] New algorithm
- [ ] Update to an existing algorithm
- [ ] Error
- [ ] Proposal to the Repository

**Details:**
Add State Design Pattern in Java, to Design Patterns.

**Comment by ighosh98** [comment]
Can I begin work on this issue?

**Comment by AdiChat** [comment]
@ighosh98 Yes, you can get started working on this and open [a pull request](https://iq.opengenus.org/create-a-pull-request-at-github/) for review 👍

**Comment by ighosh98** [comment]
Please review #5945
